nixos/modules/services/web-servers/nginx/vhost-options.nix

```diff
@@ -119,13 +119,15 @@ with lib;
     };
 
     sslCertificate = mkOption {
-      type = types.path;
+      type = types.nullOr types.path;
+      default = null;
       example = "/var/host.cert";
       description = "Path to server SSL certificate.";
```

Comment:
The description need an update.

Reply:
Do they? `sslTrustedCertificate` is nullable and its description doesn't mention it.

Reply:
Doesn't that moreso imply `sslTrustedCertificate` also requires an update to the description? 😄

Reply:
Should I also update the descriptions for `root`, `globalRedirect`, `basicAuthFile`... ?

I don't understand why the description should mention what the type already says. It's not like the behaviour when the option is null is surprising.

Reply:
I think the point here was that your configuration isn't the norm and could be a cause of confusion. Added documentation would clarify.

And speaking of clarification, I should clarify my prior comment: I was making a bit of a joke that existing nixos code isn't always the best so that shouldn't be your only justification. Maybe that nixos code was written long ago and could be improved. I didn't even look at the option to validate if my comment was valid or not, it was just to demonstrate a point.

Reply:
Point taken. Would something like "This can be left unset, for example if `ssl_reject_handshake` is used." do?

Reply:
I think that's a good description.

Reply:
> This can be nulled to unset the option which is required for ssl_reject_handshake.